docs/source/contributing.rst

```diff
@@ -0,0 +1,283 @@
+
+- Install either `Anaconda <http://docs.continuum.io/anaconda/>`_ or
+  `miniconda <http://conda.pydata.org/miniconda.html>`_
+- Make sure that you have :ref:`cloned the repository <contributing.forking>`
+- ``cd`` to the *xyzspaces** source directory
+
+Tell conda to create a new environment, named ``xyz_dev``, or any other name you would like
+for this environment, by running::
+
+      conda create -n xyz_dev python
+
+This will create the new environment, and not touch any of your existing environments,
+nor any existing python installation.
+
+To work in this environment, you need to ``activate`` it. The instructions below
+should work for both Windows, Mac and Linux::
+
+      conda activate xyz_dev
+
+Once your environment is activated, you will see a confirmation message to
+indicate you are in the new development environment.
+
+To view your environments::
+
+      conda info -e
+
+To return to you home root environment::
+
+      conda deactivate
+
+See the full conda docs `here <http://conda.pydata.org/docs>`__.
+
+At this point you can easily do a *development* install, as detailed in the next sections.
+
+3) Installing Dependencies
+--------------------------
+
+To run *xyzspaces* in an development environment, you must first install
+*xyzspaces*'s dependencies. We suggest doing so using the following commands
+(executed after your development environment has been activated)::
+
+    pip install -r requirements.txt
+    pip install -r requirements_dev.txt
+
+This should install all necessary dependencies.
+
+
+4) Making changes and writing tests
+-------------------------------------
+
+*xyzspaces* is serious about testing and strongly encourages contributors to embrace
+`test-driven development (TDD) <http://en.wikipedia.org/wiki/Test-driven_development>`_.
+This development process "relies on the repetition of a very short development cycle:
+first the developer writes an (initially failing) automated test case that defines a desired
+improvement or new function, then produces the minimum amount of code to pass that test."
+So, before actually writing any code, you should write your tests.  Often the test can be
+taken from the original GitHub issue.  However, it is always worth considering additional
+use cases and writing corresponding tests.
+
+Adding tests is one of the most common requests after code is pushed to *xyzspaces*.  Therefore,
+it is worth getting in the habit of writing tests ahead of time so this is never an issue.
+
+*xyzspaces* uses the `pytest framework <http://doc.pytest.org/en/latest/>`_.
+
+Writing tests
+~~~~~~~~~~~~~
+
+All tests should go into the ``tests`` directory. This folder contains many
+current examples of tests, and we suggest looking to these for inspiration.
+
+Running the test suite
+~~~~~~~~~~~~~~~~~~~~~~
+
+The tests can then be run directly inside your Git clone (without having to
+install *xyzspaces*) by typing::
+
+    pytest -v --cov=xyzspaces tests
+
+5) Run liniting
```

Comment:
Typo?

Reply:
Do you mean linting? I think it is correct.

Reply:
I guess it should be linting...

Reply:
sorry, my bad. fixed typo.